common/src/main/java/org/opensearch/ml/common/agent/MLToolSpec.java

```diff
@@ -24,20 +26,31 @@
 @EqualsAndHashCode
 @Getter
 public class MLToolSpec implements ToXContentObject {
+    public static final Version MINIMAL_SUPPORTED_VERSION_FOR_TOOL_CONFIG = CommonValue.VERSION_2_17_0;
```

Comment:
Isn't 2.17.0 already released so this would be a 2.18.0 feature?

Reply:
We want to catch up with 2.17 patch.

Reply:
2.17.0 open source is already released.  If you want to change the version on the back port to that branch that gets synced elsewhere that’s a reasonable option but for anyone consuming the maven central artifact BWC would break with that version check.

Reply:
Actually looks like a 2.17.1 release is happening so it should be changed to that.

Reply:
Except they don't want any new features, just critical bugfixes. So nevermind :|

Reply:
This PR is not a security fix. We can leave it to 2.18.

Reply:
Based on the discussion with Yuye, I will make the version 2.18. Thanks @yuye-aws